From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Eneas U de Queiroz <cote2004-github@yahoo.com>
Date: Thu, 27 Sep 2018 08:30:24 -0300
Subject: Add librecmc targets

Targets are named: linux-$(CONFIG_ARCH)-librecmc

Signed-off-by: Eneas U de Queiroz <cote2004-github@yahoo.com>

--- /dev/null
+++ b/Configurations/25-librecmc.conf
@@ -0,0 +1,56 @@
+## Openwrt "CONFIG_ARCH" matching targets.
+
+# The targets need to end in '-librecmc' for the AFALG patch to work
+
+my %targets = (
+    "librecmc" => {
+	template	=> 1,
+	CFLAGS		=> add("\$(LIBRECMC_OPTIMIZATION_FLAGS)"),
+    },
+    "linux-aarch64-librecmc" => {
+        inherit_from    => [ "linux-aarch64", "librecmc" ],
+    },
+    "linux-arc-librecmc" => {
+        inherit_from    => [ "linux-latomic", "librecmc" ],
+    },
+    "linux-arm-librecmc" => {
+        inherit_from    => [ "linux-armv4", "librecmc" ],
+    },
+    "linux-armeb-librecmc" => {
+        inherit_from    => [ "linux-armv4", "librecmc" ],
+    },
+    "linux-i386-librecmc" => {
+        inherit_from    => [ "linux-x86", "librecmc" ],
+    },
+    "linux-mips-librecmc" => {
+        inherit_from    => [ "linux-mips32", "librecmc" ],
+    },
+    "linux-mips64-librecmc" => {
+        inherit_from    => [ "linux64-mips64", "librecmc" ],
+    },
+    "linux-mips64el-librecmc" => {
+        inherit_from    => [ "linux64-mips64", "librecmc" ],
+    },
+    "linux-mipsel-librecmc" => {
+        inherit_from    => [ "linux-mips32", "librecmc" ],
+    },
+    "linux-powerpc-librecmc" => {
+        inherit_from    => [ "linux-ppc", "librecmc" ],
+    },
+    "linux-powerpc64-librecmc" => {
+        inherit_from    => [ "linux-ppc64", "librecmc" ],
+        perlasm_scheme  => "linux64v2",
+    },
+    "linux-riscv64-librecmc" => {
+        inherit_from    => [ "linux-generic64", "librecmc" ],
+        perlasm_scheme   => "linux64",
+    },
+    "linux-x86_64-librecmc" => {
+        inherit_from    => [ "linux-x86_64", "librecmc" ],
+    },
+
+### Basic default option
+    "linux-generic32-librecmc" => {
+        inherit_from    => [ "linux-generic32", "librecmc" ],
+    },
+);
